include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_histogram.h

```diff
@@ -285,9 +285,9 @@ struct __histogram_general_registers_local_reduction_submitter<__iters_per_work_
 template <::std::uint16_t __iters_per_work_item, ::std::uint8_t __bins_per_work_item, typename _ExecutionPolicy,
           typename _Range1, typename _Range2, typename _BinHashMgr>
 auto
-__histogram_general_registers_local_reduction(_ExecutionPolicy&& __exec, const sycl::event& __init_event,
-                                              ::std::uint16_t __work_group_size, _Range1&& __input, _Range2&& __bins,
-                                              const _BinHashMgr& __binhash_manager)
+__histogram_general_registers_local_reduction(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec,
```

Comment:
What is the purpose of adding the tag to the signature of internal implementation details beyond all the "decisions" which the tag participates?  (general question, not specific to histogram)

For histogram in particular, it seems that any function past `__parallel_histogram` has already made all the decisions it will make based on tag, and there is no alternative to `oneapi::dpl::__internal::__device_backend_tag`.

Reply:
Or is there some decision that the tag extends as far as the policy extends?

Reply:
I have added `oneapi::dpl::__internal::__device_backend_tag` here to have guaranties that this code will work only for device/fpga backends.

Reply:
About alternatives - we also have
```C++
#if _ONEDPL_FPGA_DEVICE
struct __fpga_backend_tag : __device_backend_tag
{
};
#endif
```

Reply:
Yes, I suppose there is a chance in the future we may want to specialize individual details based on FPGA vs general device tags. This is not a blocker for me, but in practice it does add an extra parameter to already long signature.  

One could argue that we could omit this beyond current existing "decisions" the tag makes, and if we decide to specialize for FPGA in the future, we could then easily add it to the necessary calls to provide the tag information to the specialization point.  Just raising the point, but I'll leave it up to you.